activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -683,6 +683,10 @@ def extract_precision(sql_type)
       EMULATE_BOOLEANS_TRUE = { emulate_booleans: true }.freeze
 
       private
+        def nonbinary_data_type?(type)
+          TYPE_MAP.lookup(type).is_a?(Type::String) || TYPE_MAP.lookup(type).is_a?(Type::Text)
+        end
```

Comment:
Would make more sense to call this method `text_type?` or `collection_type?`

Reply:
I was thinking the same thing this morning. I'll make the change!